.github/workflows/ci-ubuntu.yml

```diff
@@ -112,8 +112,7 @@ jobs:
 ## INSTALLATION STEPS
 ########################################################################
 
-      - name: Install cabal
-        if: steps.cache-cabal.outputs.cache-hit != 'true'
+      - name: Install ghc & cabal
```

Comment:
Hmm this feels like we're throwing the baby out with the bath water. 99% of the time the cache hit will be good and we'll save a lot of work right? Is the syntax expressive enough that we can write something like:
```
if: steps.cache-cabal.outputs.cache-hit != 'true' or run `ghc --version` != ${{ env.GHC_VERSION }}
```
No worries if not!

Reply:
We still keep the cache, we just insist on downloading & installing the specific ghc & cabal no matter what.
We should not be recompiling anything further down the action as long as the cache hits.

Reply:
Okay, I was more thinking about the expense of installing GHC and cabal themselves, but if its negligible then :+1: